src/plugins/intel_cpu/src/mkldnn_node.cpp

```diff
@@ -1233,8 +1233,7 @@ MKLDNNNode* MKLDNNNode::NodesFactory::create(const std::shared_ptr<ngraph::Node>
 
 bool MKLDNNNode::canBePerformedAsScaleShift(const MKLDNNNode *parentNode) const {
     size_t fusingPort = 0;
-    // @todo graph optimizer can provide parentNode as nullptr. Should be avoided
-    const size_t channelAxis = parentNode ? parentNode->getFusingAxis() : MKLDNNNode::getFusingAxis();
+    const size_t channelAxis = parentNode->getFusingAxis();
```

Comment:
I am not sure about the history of this function
But it seems there is no point to handle nullptr anymore (we have no such use case in the code)
And I assume that the only reason it was here, was this particular graph optimization, which could pass nullptr in some situations (which is not the case anymore).
Regarding the check, I don't think we should do it for this particular function.
We pass shared_ptrs and shared_ptr& all over the place and almost never check for nullptr.
The best way to handle this would be to insert "assert()" calls at the beginning of such functions.

For this particular case I propose to completely remove nullptr handling from the function.
I mean, for example, the for loop below.

Reply:
"We pass shared_ptrs and shared_ptr& all over the place and almost never check for nullptr." which is not very good for safety reasons. If we do not expect nullptr it is better to use const reference.

Reply:
Just to be clear, I totally agree with this, but don't like to make a change which introduce "alternative" handling (in this case to handle or not to handle nullptr) to the code.
So when you look at one function - it has nullptr checks, and look at the other - it does not have such check.
Immediate question arises for the reader of the code - why is it so.
At the moment we kind of have implicit agreement that if function does not handle nullptr then it does not expect it at all and there is no such use cases.
Maybe we could introduce a rule / development practice to always insert "assert()" calls or something like this.

Reply:
Let's add IE_ASSERT with null check for `parentNode` just to make our life a little easier in case we got nullptr due to some bug. Regarding the loop below it looks even more strange since `getParent()` perform the mull check internally though it is not documented it is widely used and moreover I can not get why a raw pointer is used there.

Reply:
My assumption is that the reason - there is a use case when _this_ is passed as an argument.